### DISCUSSION TITLE
Ubuntu 20.04 permission problem

### DISCUSSION THREAD

**Discussion by MrTheBarbarian** [Koenkk/zigbee2mqtt#13408] [General]
Hi there!
After update of ubuntu, m2z doesn't start. I got permission problem of the USB: 
`  Error: Error while opening serialport 'Error: Error: Permission denied, cannot open /dev/ttyUSB0'`

i found when I run this: sudo chmod 666 /dev/ttyUSB0 , m2z starts normally.
So, the problem is that, I have to do this every time when the system has rebooted.
Any suggestions will be appreciated!
Thanks!

**Comment by Nardol** [comment]
What is the output of `sudo ls -l /dev/ttyUSB0`

**Reply by MrTheBarbarian** [reply]
crw-rw---- 1 root dialout 188, 0 Aug  6 08:58 /dev/ttyUSB0

**Reply by Nardol** [reply]
You have [multiple solutions](https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html#verify-that-the-user-you-run-zigbee2mqtt-as-has-write-access-to-the-port).
The quickest and easiest one is to type the following command, where `$USER` has to be replaced by the user running Zigbee2MQTT if the user you are logged in to run it is different than the one who runs Zigbee2MQTT:
`sudo usermod -a -G dialout $USER`
Then Zigbee2MQTT will be able to access the serial port.

**Reply by MrTheBarbarian** [reply]
Hi. I tried it, but it doesn't work. It's the same. Could you please suggest something else?

**Comment by Nardol** [comment]
Have you also tried [other solution from the documentation](https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html#verify-that-the-user-you-run-zigbee2mqtt-as-has-write-access-to-the-port)?

**Reply by MrTheBarbarian** [reply]
Оf course! :-)

**Reply by Nardol** [reply]
With the UDev solution, you also have to change the port in your Zigbee2MQTT configuration but I suppose you tried within doing it :-)
Have you tried to use `/dev/serial/by-id/id_of_your_zigbee_stick` instead of `/dev/ttyUSB0` ?
It is more recommended to use the `by-id` path, if you have something else using `ttyUSB0` which is used it would make thing more sure.
I don't see why it would fix the issue but if you have tried the two solutions I don't see anything else than suggest that.

And what exactly have you updated on Ubuntu? Changing Ubuntu version completely?

**Reply by MrTheBarbarian** [reply]
Hi! I tried and it is not works.

**Comment by Nardol** [comment]
If you run Zigbee2MQTT using SystemD, add `ExecStartPre=chmod 666 /dev/ttyUSB0` in the `[Service]` section, for example above the line beginning with `ExecStart`

It is not a real solution because it is only automating what you have to do on each reboot but it should work and without more information about your installation and what you did exactly before it began to stop working I don't see any other solution.

**Reply by MrTheBarbarian** [reply]
Thank you for the reply! I made something similar. I put this command in crontab. It's works.

**Comment by kalbfuss** [comment]
Is the user you are using to run z2m a member of the dialout group?

**Reply by MrTheBarbarian** [reply]
Sorry. I not sure, what are you thinking about. I do not know to check it. I'm run z2m as explained on official site.

**Reply by kalbfuss** [reply]
Does that mean you run z2m in a docker container?
What is the owning group of /dev/ttyUSB0 in your Ubuntu system?

**Reply by kalbfuss** [reply]
You may want to check this one: https://stackoverflow.com/questions/68997623/permission-denied-when-accessing-ttyusb-in-a-privileged-docker-container-using
